File: app/src/main/java/ch/bailu/aat/gpx/parser/IntegerScanner.java
package ch.bailu.aat.gpx.parser;
import java.io.IOException;

import ch.bailu.simpleio.io.SimpleStream;


public class IntegerScanner {
	private int integer;
	private final SimpleStream stream;
	
	public IntegerScanner(SimpleStream s) {
		stream = s;
	}
	
	
	public void scan() throws IOException {
		integer=0;
		
		stream.read();
		stream.skipWhitespace();
		
		while (true) {
			if (stream.haveDigit()) {
				integer*=10;
				integer+=stream.getDigit();
			} else {
				break;
			}
			stream.read();
		}
		
	}
	
	public int getInteger() {
		return integer;
	}
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/file/xml/parser/util/IntegerScanner.kt
package ch.bailu.aat_lib.file.xml.parser.util

import java.io.IOException

class IntegerScanner : AbsScanner() {
    var integer = 0
        private set

    @Throws(IOException::class)
    override fun scan(stream: Stream) {
        integer = 0
        stream.read()
        stream.skipWhitespace()
        while (true) {
            if (stream.haveDigit()) {
                integer *= 10
                integer += stream.digit
            } else {
                break
            }
            stream.read()
        }
    }
}
